docs/endpoint.md

```diff
@@ -5,14 +5,132 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
+```
+
+Create a workspace
+```
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Setup the wp77xx target profile
+```
+$ leaf setup legato-latest -p swi-wp77_3.3.1
+```
+
+Finally, make the wp77xx endpoint target. Be careful with the directory of tangle-accelerator. It should be located within the workspace directory.
+```
+$ git clone https://github.com/DLTcollab/tangle-accelerator.git
+$ cd tangle-accelerator
+$ make TARGET=wp77xx legato # build endpoint as wp77xx target
+$ make TESTS=true TARGET=wp77xx legato # build endpoint as wp77xx target in test mode
+```
+
+### How to build endpoint application for native target
+
+Install required packages:
+```
+$ sudo apt-get install -y   \
+    autoconf                \
+    automake                \
+    bash                    \
+    bc                      \
+    bison                   \
+    bsdiff                  \
+    build-essential         \
+    chrpath                 \
+    cmake                   \
+    cpio                    \
+    diffstat                \
+    flex                    \
+    gawk                    \
+    gcovr                   \
+    git                     \
+    gperf                   \
+    iputils-ping            \
+    libbz2-dev              \
+    libcurl4-gnutls-dev     \
+    libncurses5-dev         \
+    libncursesw5-dev        \
+    libsdl-dev              \
+    libssl-dev              \
+    libtool                 \
+    libxml2-utils           \
+    ninja-build             \
+    python                  \
+    python-git              \
+    python-jinja2           \
+    python-pkg-resources    \
+    python3                 \
+    texinfo                 \
+    unzip                   \
+    wget                    \
+    zlib1g-dev
+```
+
+Create a workspace
+```
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Download legato by [git-repo](https://gerrit.googlesource.com/git-repo)
+```
+$ repo init -u git://github.com/legatoproject/manifest
+$ repo sync
+```
+
+You can also clone a specific version of `legato`
+```
+$ repo init -u git://github.com/legatoproject/manifest -m legato/releases/19.11.1.xml # specific legato 19.11.1 version
+$ repo sync
+```
+
+Build legato as native target
 ```
-$ bazel build --define https=enable //endpoint:wp7702
+$ cd legato
+$ make localhost
+```
+
+Checkout the shell to legato shell
+```
+$ source ./framework/tools/scripts/configlegatoenv
+```
+
+Finally, go back to the root directory of tangle-accelerator. Use GNU Make to build the endpoint application.
+```
+$ make legato # build endpoint as native target
+```
+
+The endpoint will be built at `endpoint/_build_endpoint/localhost/app/endpoint/staging/read-only/bin/endpoint`
+
+## HTTPS Connection Support
+The endpoint uses http connection as default. The message sent to tangle-accelerator has been encrypted. So the HTTP connection would not be unsafe. To build with https connection support, add `ENABLE_HTTPS=true` option.
```

Comment:
The endpoint uses HTTP connection as default. The message sent to tangle-accelerator has been encrypted, so the HTTP connection would not be unsafe. To build with HTTPS connection support, add `ENABLE_HTTPS=true` option as following.